src/CoBuster/CoBuster/Helpers/VitalSignsProcessing.cs

```diff
@@ -78,15 +78,14 @@ public int HRProcesssing(byte[] data, Size size)
 			{
 				return -1;
 			}
-
 			
 			int width = (int)size.Width;
 			int height = (int)size.Height;
 			double RedAvg;
 			double GreenAvg;
 
-			GreenAvg = ImageProcessing.DecodeYUV420SPtoRedBlueGreenAvg(data, height, width, 3); //1 stands for red intensity, 2 for blue, 3 for green
-			RedAvg = ImageProcessing.DecodeYUV420SPtoRedBlueGreenAvg(data, height, width, 1); //1 stands for red intensity, 2 for blue, 3 for green
+			GreenAvg = ImageProcessing.DecodeYUV420SPtoRedBlueGreenAvg(ImageProcessing.Clone(data), height, width, 3); //1 stands for red intensity, 2 for blue, 3 for green
```

Comment:
Curious.  Why the need for the clone array function and using it during image processing?